### DISCUSSION TITLE
セキュリティ通知の調査

### DISCUSSION THREAD

**Discussion by kazzyfrog** [lgtm-factory/lgtm-factory#128] [Scrap]
GitHubのセキュリティアラートが、１つ点いているので、調査します！

おそらく、husky or shaden/ui の導入あたりで、混入した？感じですかね。

https://github.com/lgtm-factory/lgtm-factory/security/dependabot/5

<img width="148" alt="スクリーンショット 2024-08-26 17 47 00" src="https://github.com/user-attachments/assets/bbae4d9b-842e-44d5-988a-8b6b82774983">

**Comment by kazzyfrog** [comment]
あれ、というか上記のページって皆さんアクセスできるのかな。。？

**Reply by kagomen** [reply]
アクセスできます！

**Reply by kazzyfrog** [reply]
承知です！！👍

**Comment by kagomen** [comment]
以下で、心当たりあります！

https://github.com/lgtm-factory/lgtm-factory/pull/115/commits/65ddf1eec91ff11a8a3270142767fd5fb72c4c3b

完全には記憶してないんですが、多分、なぜかpackage-lock.jsonでコンフリクトを起こしてて、
それを解消するために一旦node_modulesとpackage-lock.jsonを削除して、npm installし直して、コンフリクト解消されていたので `git push`した、という流れだったはずです

ここで、何かが混入した……？

**Reply by kazzyfrog** [reply]
なるほど！
一旦問題なさそうなので、時間がある時に、調査してみます！！

**Reply by kagomen** [reply]
すみませんー💦
自分も頭の片隅に置いておきます！

**Comment by kagomen** [comment]
`package.json`を変更せずに`package-lock.json`が変更されることはあります。以下のようなケースで発生する可能性があります：

1. 依存関係のアップデート:
   `npm update`コマンドを実行すると、`package.json`で指定されたバージョン範囲内で依存パッケージが更新され、`package-lock.json`が変更される可能性があります。

2. 新しい環境でのインストール:
   異なるOSや環境で`npm install`を実行すると、プラットフォーム固有の依存関係が変更される可能性があります。

3. npmのバージョン変更:
   npm自体のバージョンが変更されると、`package-lock.json`のフォーマットや内容が変更される場合があります。

4. 間接的な依存関係の変更:
   直接インストールしていないパッケージ（依存パッケージの依存関係）がアップデートされた場合、`package-lock.json`が変更されることがあります。

5. インテグリティハッシュの再計算:
   何らかの理由でパッケージのインテグリティハッシュが再計算された場合、`package-lock.json`が更新されることがあります。

6. npmレジストリの変更:
   パッケージのメタデータがnpmレジストリで更新された場合、`package-lock.json`に反映されることがあります。

これらの変更は通常、プロジェクトの動作に大きな影響を与えることはありませんが、バージョン管理システムを使用している場合は注意が必要です。チーム開発では、`package-lock.json`の変更をコミットし、他のメンバーと同期することが重要です。

**Comment by kagomen** [comment]
めも

<img width="682" alt="スクリーンショット 2024-08-27 3 33 36" src="https://github.com/user-attachments/assets/40ecd023-4993-40f4-bbd7-f2f6b699d0a4">

**Comment by kagomen** [comment]
memo

https://engineering.mobalab.net/2019/08/08/package-json%E3%81%A8package-lock-json%E3%81%AE%E9%81%8B%E7%94%A8%E6%96%B9%E6%B3%95%E3%81%AB%E3%81%A4%E3%81%84%E3%81%A6/

**Comment by kazzyfrog** [comment]
Dependabot の自動アップデートボタンを押して、
自動で脆弱性のあるパッケージをアップデートしてPRが作成しました！

- #143 
一応ビルドは成功しているので、
少し触ってみて正常に動くようだったら、マージしていいかもですね！

**Reply by kagomen** [reply]
そんな機能が！

**Comment by kagomen** [comment]
package.jsonさえ消さなければ問題ないと思ってpackage-lock.jsonを気軽に消してしまったけど、そういう感じのファイルではなさそう・・・？
気をつけます……！🙏

https://zenn.dev/nekoya/articles/c6057fbb896391

> 直接的に利用しているライブラリに脆弱性が見つかった場合などは、lockfileではなくpackage.jsonでのバージョン指定を問題ない状態に保ちましょう。
そもそもpackage.jsonでバージョンを指定しているのに、そこでの指定を不完全にして更に別のファイルに依存する必要はありません。
package-lock.jsonなどのlockfileは間接的に依存するライブラリのバージョンを管理するためのものと考えるのがいいでしょう。

**Reply by kazzyfrog** [reply]
全然大丈夫ですよ！
コードレビューの際も、`package-lock.json`ってなんか隠れているので、見づらいんですよね。。！
